lib_eio/buf_write.mli

```diff
@@ -123,6 +123,19 @@ val cstruct : t -> Cstruct.t -> unit
     It is safe to modify [cs] after this call returns.
     For large cstructs, it may be more efficient to use {!schedule_cstruct}. *)
 
+val printf : t -> ('a, Format.formatter, unit) format -> 'a
+(** [printf t fmt x y z] formats the arguments according to the format string [fmt].
+    It supports all formatting and pretty-printing features of the Format module.
+    Accordingly, explicit flushes using [@.] or [%!] must perform a full (blocking) flush
+    so consider using [Fiber.fork] in such cases. *)
```

Comment:
Telling people to use Fiber.fork as a work-around seems odd, and might lead to errors if two fibers use the same formatter at once. Possibly the flushing behaviour should be controlled by an argument here.

Also, the word `must` seems out of place. We're not asking the user to do anything, just noting that we have chosen to do a full flush in this case.

Reply:
Fixed. Agreed. I've got reservations about an argument for the flushing behavior, we can talk more about it at the next dev meeting.